webapp/src/components/modals/create_issue/create_issue.jsx

```diff
@@ -51,6 +53,11 @@ export default class CreateIssueModal extends PureComponent {
             const fields = {...this.state.fields};
             fields.description = this.props.post.message;
             this.setState({fields}); //eslint-disable-line react/no-did-update-set-state
+        } else if (this.props.channelId && (this.props.channelId !== prevProps.channelId || this.props.description !== prevProps.description)) {
```

Comment:
This is getting unruly. Can we derive the description in the mapStateToProps function and just pass it correctly? Then we can simplify this. 

I would like to get rid of the derived state here. But I am not sure how.

Reply:
Like we chatted about over zoom, this will be split into a separate refactoring PR. 
I'd love to move all this derived state into props, and then just letting the component update itself when the state changes.